File: TransCoder_tokenized_test_set_functions/python/MAXIMUM_NUMBER_CHARACTERS_TWO_CHARACTER_STRING_1.py
def maximumChars ( str1 ) :
    n = len ( str1 )
    res = - 1
    firstInd = [ - 1 for i in range ( MAX_CHAR ) ]
    for i in range ( n ) :
        first_ind = firstInd [ ord ( str1 [ i ] ) ]
        if ( first_ind == - 1 ) :
            firstInd [ ord ( str1 [ i ] ) ] = i
        else :
            res = max ( res, abs ( i - first_ind - 1 ) )
    return res

File: TransCoder_tokenized_test_set_functions/java/MAXIMUM_NUMBER_CHARACTERS_TWO_CHARACTER_STRING_1.java
static int maximumChars ( String str ) {
  int n = str . length ( );
  int res = - 1;
  int [ ] firstInd = new int [ MAX_CHAR ];
  for ( int i = 0;
  i < MAX_CHAR;
  i ++ ) firstInd [ i ] = - 1;
  for ( int i = 0;
  i < n;
  i ++ ) {
    int first_ind = firstInd [ str . charAt ( i ) ];
    if ( first_ind == - 1 ) firstInd [ str . charAt ( i ) ] = i;
    else res = Math . max ( res, Math . abs ( i - first_ind - 1 ) );
  }
  return res;
}